src/rust/engine/fs/src/posixfs_test.rs

```diff
@@ -0,0 +1,474 @@
+use tempfile;
```

Comment:
This file should be named `posixfs_tests.rs` (the `s` at the end)

This makes me think, could it be worth adding a pre-commit hook that checks that no files are named (`_test.rs`)?